src/jdk.httpserver/share/classes/sun/net/httpserver/FixedLengthOutputStream.java

```diff
@@ -64,6 +64,9 @@ public void write (int b) throws IOException {
         }
         out.write(b);
         remaining --;
+        if(remaining==0) {
+            close();
```

Comment:
`close()` has the side effect of closing the input stream. This again may break other code.

Reply:
We can remove this. As I stated, it is a bad bug and will leave the connection in a corrupted state if the handler does not call close() on the outstream or the exchange, so I don't think existing code "works".

Also, I think issuing the close() here is benign, because any further writes would throw an exception (too much data), and a subsequent close (or multiple closes) are ignored.

Essentially, once you have written all of the data you stated you would send, it is a critical error if you do anything on the output stream other than flush() or close(). I added a check that a flush() on a closed stream is ignored.

Reply:
as to the input stream being closed, that already would have happened. If the handler closed the output stream, then tried to read the input stream, it would fail.

in retrospect, I think this should be removed, because if the handler was not closing the output stream before, the connection would have been dead, so it seems doubtful this was the case in the field. and then we can fix the broken tests to not not have a bug in not properly closing the output stream.